modules/api/src/main/java/org/apache/fluo/api/data/Bytes.java

```diff
@@ -355,6 +354,36 @@ public static final Bytes of(String s, Charset c) {
   }
 
   /**
+   * Checks if this has the passed prefix
+   * 
+   * @param prefix
+   * @return true or false
+   */
+  public boolean startsWith(Bytes prefix) {
+    for (int i = 0; i < prefix.length; i++) {
+      if (this.byteAt(i) != prefix.byteAt(i))
+        return false;
+    }
+    return true;
+  }
+
+  /**
+   * Checks if this has the passed suffix
+   * 
+   * @param suffix
+   * @return true or false
+   */
+  public boolean endsWith(Bytes suffix) {
+    int startOffset = this.length - suffix.length;
+    for (int i = startOffset; i < this.length; i++) {
+
+      if (this.byteAt(i) != suffix.byteAt(i - startOffset))
```

Comment:
Looking at this code prompted me to open #826.  Would be ok to make these improvements for endsWith and startsWith in this PR or in a later PR for #826 that improves all the code.